src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -290,7 +290,10 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
             if (bestItem != null)
             {
                 selectedItemIndex = itemsInList.IndexOf(i => Equals(i.FilterResult.CompletionItem, bestItem));
-                if (selectedItemIndex > -1 && bestItem != null && matchingItems.Length == 1 && filterText.Length > 0)
+                if (selectedItemIndex > -1 &&
+                    bestItem != null &&
+                    matchingItems.Where(r => !r.DisplayText.StartsWith("★")).Count() == 1 &&
```

Comment:
i really really really don't like hardcoding this in.  I thought i opened a PR a long time ago to properly represent intellicode items so we didn't have to hack things in this way.  @dpoeschl Didn't you review/merge that in?

Reply:
Yeah, this was added:

```c#
         /// <summary>
        /// An optional prefix to be displayed prepended to <see cref="DisplayText"/>. Can be null.
        /// Pattern-matching of user input will not be performed against this, but only against <see
        /// cref="DisplayText"/>.
        /// </summary>
        public string DisplayTextPrefix { get; }
```

Reply:
Unfortunately, those are items not provided by Roslyn but provided by IntelliCode. They seem to put stars into displayText. No matter: the right fix is to consume a special property for those items not to review they texts or prefixes. When @AmadeusW will extend the Editor's API with those new properties, we will replace this temporarily solution.

Reply:
> Unfortunately, those are items not provided by Roslyn but provided by IntelliCode. They seem to put stars into displayText. 

Blargh, the purpose of my pr (and it looks like this must have gotten lost in the communication) was for the intellicode hacks to stop and for them to use the real API here to pass this information in.  

Who can we talk to on their end to get that to happen?

Reply:
I think that now the Editor team has to work with partners to advocate the right approach of using the data structure. @AmadeusW can help.

For now, we are applying a temporarily workaround for 16.1. In 16.2, we should be able to consume a proper field not a hack.

If Roslyn can be called for filtering other providers items, the Editor and other providers will still be dependent on Roslyn rules defined for data.

Reply:
@ivanbasov I'm not quire sure hte "Editor" part of this discussion.  Isn't this all purely at the Roslyn layer? These are Roslyn completoin items, which intellicode is creating.  THye hacked stuff in, but we have an appropriate API they should be using.  The conversation was supposed to happen with them to move to that API.  But it doesn't seem to have happened.  I'm curious who is driving htat so that roslyn doesn't need to have more and more hacks added to it.

--

As an aside, continually adding hacks, and not going to clean them up, is something i've seem so many times in my work, and it's such a drag and on future agility.  I get that there are market/team pressures that cause this.  but if it is allowed to just continually happen, without actual cleanup, you get into a situation where the costs just keep building and end up forcing the burn-down-and-start-over approach.